agent/api/task/service_connect.go

```diff
@@ -65,3 +81,246 @@ type DNSConfigEntry struct {
+			return fmt.Errorf("service connect config validation: no service connect DNS config")
+		}
+	}
+	for _, dnsEntry := range scDnsConfligList {
+		// HostName is a required field
+		if dnsEntry.HostName == "" {
+			return fmt.Errorf("service connect config validation: no service connect DNS config host name")
+		}
+		// Address is a required field
+		if dnsEntry.Address == "" {
+			return fmt.Errorf("service connect config validation: no service connect DNS config address")
+		}
+	}
+	return nil
+}
+
+// validateStatsConfig validates the service connect stats config path exists.
+func validateStatsConfig(scStatsConfig StatsConfig) error {
+	// stats config is a required field
+	if scStatsConfig.Path == "" {
+		return fmt.Errorf("service connect config validation: no service connect stats config")
+	}
+	return nil
+}
+
+// validateContainerName validates the service connect container name exists, and it exists in the task definition as well.
+func validateContainerName(scContainerName string, taskContainers []*ecsacs.Container) error {
+	// container name is a required field
+	if scContainerName == "" {
+		return fmt.Errorf("service connect config validation: no service connect container name")
+	}
+	// validate the specified service connect container name exists in task
+	for _, container := range taskContainers {
+		if aws.StringValue(container.Name) == scContainerName {
+			return nil
+		}
+	}
+	return fmt.Errorf("service connect config validation: service connect container name: %s does not exist in task", scContainerName)
+}
+
+// validateCIDR validates the passed CIDR is valid.
+func validateCIDR(cidr, version string) error {
+	_, _, err := net.ParseCIDR(cidr)
+	if err != nil {
+		return fmt.Errorf("service connect config validation: %s: %s is invalid", version, cidr)
+	}
+	return nil
+}
+
+// validateEgressConfig validates the service connect egress config has listener name and vip if egress config is not empty.
+func validateEgressConfig(scEgressConfig EgressConfig) error {
+	// EgressConfig can be empty for the first service since there are no other tasks that it can talk to
+	if scEgressConfig.ListenerName == "" && scEgressConfig.VIP.IPV4CIDR == "" && scEgressConfig.VIP.IPV6CIDR == "" {
+		return nil
+	}
+	// ListenerName is a required field if egress config is not empty
+	if scEgressConfig.ListenerName == "" {
+		return fmt.Errorf("service connect config validation: no service connect listener name")
+	}
+	// VIP is a required field if egress config is not empty
+	if scEgressConfig.VIP.IPV4CIDR == "" && scEgressConfig.VIP.IPV6CIDR == "" {
+		return fmt.Errorf("service connect config validation: no service connect VIP")
+	}
+	// validate IPV4CIDR if it exists
+	if scEgressConfig.VIP.IPV4CIDR != "" {
+		trimmedIpv4cidr := strings.TrimSpace(scEgressConfig.VIP.IPV4CIDR)
+		if err := validateCIDR(trimmedIpv4cidr, "IPV4CIDR"); err != nil {
+			return err
+		}
+	}
+	// validate IPV6CIDR if it exists
+	if scEgressConfig.VIP.IPV6CIDR != "" {
```

Comment:
Shall we add a validation that IPV6CIDR must not be empty when the task is IPV6 enabled?

Reply:
Validation and its test are added in commit 3d8d95406e74fc25d3c00d384a11cd5cd4ae55f9. Please take a look on `service_connect_validation.go` and `service_connect_validation_test.go`. Thanks.